pkg/network/validation.go

```diff
@@ -0,0 +1,85 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/nspcc-dev/neofs-api-go/pkg/netmap"
+)
+
+const (
+	// maxProtocolsAmount is maximal amount of protocols
+	// in multiaddress after parsing with network.AddressFromString
+	maxProtocolsAmount = 3
+
+	// minProtocolsAmount is minimal amount of protocols
+	// in multiaddress after parsing with network.AddressFromString:
+	// host(ip) and port.
+	minProtocolsAmount = 2
+
+	// network protocols
+	dns, ip4, ip6 = "dns4", "ip4", "ip6"
+
+	// transport protocols
+	tcp = "tcp"
+)
+
+var (
+	errIncorrectProtocolAmount         = errors.New("numbers of protocols in multiaddress incorrect")
+	errUnsupportedNetworkProtocol      = errors.New("unsupported network protocol in multiaddress")
+	errUnsupportedTransportProtocol    = errors.New("unsupported transport protocol in multiaddress")
+	errUnsupportedPresentationProtocol = errors.New("unsupported presentation protocol in multiaddress")
+)
+
+// VerifyAddress validates multiaddress of n.
+//
+// If n's address contains more than 3 protocols
+// or less than 2 protocols an error returns.
+//
+// If n's address's protocol order is incorrect
+// an error returns.
+//
+// Correct composition(and order from low to high level)
```

Comment:
`low` / `high` in what terms?

Reply:
i meant OSI model since could not decide how to explain in other words